.happy/terraform/modules/sfn/main.tf

```diff
@@ -12,196 +13,283 @@ resource "aws_sfn_state_machine" "state_machine" {
+              "Resource": "arn:aws:states:::batch:submitJob.sync",
+              "Parameters": {
+                "JobDefinition": "${var.job_definition_arn}",
+                "JobName": "validate_anndata",
+                "JobQueue.$": "$.job_queue",
+                "ContainerOverrides": {
+                  "Environment": [
+                    {
+                      "Name": "MANIFEST",
+                      "Value.$": "$.manifest"
+                    },
+                    {
+                      "Name": "DATASET_VERSION_ID",
+                      "Value.$": "$.dataset_version_id"
+                    },
+                    {
+                      "Name": "COLLECTION_VERSION_ID",
+                      "Value.$": "$.collection_version_id"
+                    },
+                    {
+                      "Name": "STEP_NAME",
+                      "Value": "validate_anndata"
+                    }
+                  ]
+                }
               },
-              {
-                "Name": "STEP_NAME",
-                "Value": "validate_anndata"
-              }
-            ]
+              "ResultPath": null,
+              "TimeoutSeconds": ${local.h5ad_timeout},
+              "Retry": [
+                {
+                  "ErrorEquals": [
+                    "AWS.Batch.TooManyRequestsException",
+                    "Batch.BatchException",
+                    "Batch.AWSBatchException"
+                  ],
+                  "IntervalSeconds": 2,
+                  "MaxAttempts": 7,
+                  "BackoffRate": 5
+                }
+              ],
+              "End": true
+            }
           }
         },
-        "ResultPath": null,
-        "TimeoutSeconds": ${local.h5ad_timeout},
-        "Retry": [ {
-            "ErrorEquals": ["AWS.Batch.TooManyRequestsException", "Batch.BatchException", "Batch.AWSBatchException"],
-            "IntervalSeconds": 2,
-            "MaxAttempts": 7,
-            "BackoffRate": 5
-        } ],
-        "Catch": [
-          {
-            "ErrorEquals": [
-              "States.ALL"
-            ],
-            "Next": "HandleErrors",
-            "ResultPath": "$.error"
-          }
-        ]
-      },
-      "AddLabels": {
-        "Type": "Task",
-        "Resource": "arn:aws:states:::batch:submitJob.sync",
-        "Next": "Cxg",
-        "Parameters": {
-          "JobDefinition": "${var.job_definition_arn}",
-          "JobName": "add_labels",
-          "JobQueue.$": "$.job_queue",
-          "ContainerOverrides": {
-            "Environment": [
-              {
-                "Name": "DATASET_VERSION_ID",
-                "Value.$": "$.dataset_version_id"
+        {
+          "StartAt": "ValidateAtac",
+          "States": {
+            "ValidateAtac": {
+              "Type": "Task",
+              "Resource": "arn:aws:states:::batch:submitJob.sync",
+              "Parameters": {
+                "JobDefinition": "${var.job_definition_arn}",
```

Comment:
Do you think it'll be necessary to create a separate batch job def for validateAtac? reminder that, with the dask changes, we changed this batch default (used for validate and add labels) to use 8 GB 1 vcpu. CXG has its own definition using 16 GB and 2 vcpu. Will validate atac need more? Is it chunked / parallelized similarly?

Reply:
yeah, I think we can track that in a follow-up issue. When I tried, I was finding that the speed-up was not as much of a performance increase as I expected (almost negligible even on larger datasets) but that might've been an implementation error on my part.